commands/management/add_mysql.go

```diff
@@ -176,6 +180,35 @@ func (cmd *addMySQLCommand) Run() (commands.Result, error) {
 		}
 	}
 
+	if cmd.DefaultsFile != "" {
+		path := cmd.DefaultsFile
+
+		type Config struct {
+			Client struct {
+				User     string
+				Password string
+			}
+		}
+
+		if strings.HasPrefix(path, "~/") {
+			usr, err := user.Current()
+			if err != nil {
+				return nil, err
+			}
+			dir := usr.HomeDir
+			path = filepath.Join(dir, path[2:])
+		}
+
+		var cfg Config
+		err := gcfg.ReadFileInto(&cfg, path)
```

Comment:
command shouldn't fail if file contains other configs. please read https://godoc.org/gopkg.in/gcfg.v1#hdr-Error_handling and https://godoc.org/gopkg.in/gcfg.v1#FatalOnly